R/oauth-init.R

```diff
@@ -10,15 +10,16 @@
 #' @export
 #' @keywords internal
 init_oauth1.0 <- function(endpoint, app, permission = NULL,
-                          is_interactive = interactive()) {
+                          is_interactive = interactive(),
+                          private_key = NULL) {
 
-  oauth_sig <- function(url, method, token = NULL, token_secret = NULL, ...) {
-    oauth_header(oauth_signature(url, method, app, token, token_secret, ...,
+  oauth_sig <- function(url, method, token = NULL, token_secret = NULL, private_key = NULL, ...) {
+    oauth_header(oauth_signature(url, method, app, token, token_secret, private_key, ...,
       callback = oauth_callback()))
   }
 
   # 1. Get an unauthorized request token
-  response <- GET(endpoint$request, oauth_sig(endpoint$request, "GET"))
```

Comment:
Given that the RFC prescriptive and the change works for four implementations in the wild, I'd say it's fine to change the default.  Let's not worry about making it optional until someone actually complains about it.